src/components/forms/SignUpForm/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { useForm } from 'react-hook-form';
+import i18next from 'i18next';
+
+import { User } from '../../../utils/types';
+import { emailRegex, passwordRegex } from '../../../constants/regex';
+
+function SignUpForm() {
+  const {
+    handleSubmit,
+    register,
+    getValues,
+    formState: { errors }
+  } = useForm<User>();
+  // eslint-disable-next-line no-console
+  const onSubmit = () => console.log(getValues());
+
+  return (
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <label className="form-label">{i18next.t('SignUp:firstName')}</label>
```

Comment:
This is a feedback rather than a comment, I sometimes create an object with the inpust and then I make a .map() something like this

```
const INPUTS_FORM = [
  {
    label: i18next.t('SignUp:firstName'),
    name: 'firstName',
    type: 'text',
    ref: {register({required: i18next.t('FormValidations:required') as string })}
  },
    {
    label: i18next.t('SignUp:firstName'),
    name: 'LastName',
    type: 'text',
    ref: {register({required: i18next.t('FormValidations:required') as string })}
  },
      {
    label: i18next.t('SignUp:firstName'),
    name: 'email',
    type: 'text',
    ref: {register({required: i18next.t('FormValidations:required') as string })}
  }
]
```


`INPUTS_FORM.map(({ label, name, ref, type }) => (
  <label className="form-label">{label}</label>
      <input
        className="form-input"
        name={name}
        type={type}
        ref={ref}
      />
))`